### DISCUSSION TITLE
Left/ Right Handed Only Mode

### DISCUSSION THREAD

**Discussion by wizard-waqas** [monkeytypegame/monkeytype#3619] [Ideas] [answered]
To better practice dedicated left/ right hand typing, create a mode that only generates words that can solely typed by the specified hand. 

For example, if the user chooses left, then generate words with the following characters only:
qwertasdfgzxcvb

And if right, then:
yuiophjklnm

**Comment by Miodec** [comment, marked as answer]
<img width="472" alt="image" src="https://user-images.githubusercontent.com/13181393/193477358-1a762e8a-5951-4d00-b828-5357eb753536.png">

You can use custom mode to filter words based on their characters / character combos. Include letters you want to type (one side) and exclude letters you don't want to type (other side)

**Reply by wizard-waqas** [reply]
I already did this. Wouldn't it be cool to have it as a feature on the most featureful typing game on the web?

**Reply by Miodec** [reply]
Im not sure what you mean "have it as a feature" when its already a feature?

**Reply by wizard-waqas** [reply]
Within the screenshot you sent me (or the screen before it), there could be a 2 buttons labeled "Left" and "Right" that automatically set the word list and filter settings accordingly, instead of having to do that stuff yourself like i did in the message below

**Reply by Miodec** [reply]
Ooh so you mean filter presets, could do that

**Reply by wizard-waqas** [reply]
Im a developer myself, but I've never contributed to an open source project. How would I get started on tackling this? So far I've cloned the Repo and found the file where the changes would be made

**Reply by Miodec** [reply]
Well, the biggest issue here is that we can't just do a "left and right" presets because different layouts have different "left and right" keys. So i think this would require dynamically building those filters based on a selected layout.

So, in the words filter popup you would need to add 2 dropdowns and a button. One dropdown to select a layout, one to select a preset (left hand, right hand, home row, top row, bottom row - those should do for now) and a button that generates the "include" and "exclude" strings and populates the inputs. Dont worry about styling things, i can touch things up after (as i usually do with most feature pulls)

If you're serious about taking this on, ill create an issue for you and assign you to it.

**Reply by wizard-waqas** [reply]
Yeah can you do it, I'd like to give it a shot

**Reply by Miodec** [reply]
Alright, comment on #3620 so i can assign you

**Comment by wizard-waqas** [comment]
For those wanting to practice right hand, heres the word list i used

```pumpkin lollipop nook moo oily poppy phony loony join mommy joy polonium junk limp onion pulp lion molly homily pull million yummy link monk holly mop ninny kiln pool oink puppy hop opinion homophony hippo pink milky monopoly moon yolo hill punky nylon you only look minimum imply```
